src/peerbook/libp2p_peerbook.erl

```diff
@@ -69,6 +69,9 @@ put(#peerbook{tid=TID, stale_time=StaleTime}=Handle, PeerList) ->
                                             NewPeerId /= ThisPeerId
                                                 andalso libp2p_peer:supersedes(NewPeer, ExistingPeer)
                                                 andalso not libp2p_peer:is_stale(NewPeer, StaleTime)
+                                                andalso (libp2p_peer:network_id(NewPeer) == libp2p_swarm:get_network_id(TID)
```

Comment:
I see this statement repeated but in the positive and negative form to check whether the network id matches. Could you extract this into a function on libp2p_peer to reduce copy-pasta?